src/assets/src/components/queue.tsx

```diff
@@ -446,14 +446,27 @@ export function QueuePage(props: PageProps<QueuePageParams>) {
         await api.removeMeeting(queue!.my_meeting!.id);
     }
     const [doLeaveQueue, leaveQueueLoading, leaveQueueError] = usePromise(leaveQueue);
-    const confirmLeaveQueue = (queueStatus: 'open' | 'closed') => {
-        const loseMessage = 'By leaving the queue, you will lose your place in line'
-        const description = 'Are you sure you want to leave the queue? ' + (
-            queueStatus === 'open'
-                ? loseMessage + '.'
-                : `The queue is currently closed. ${loseMessage} and will not be able to rejoin until the queue is reopened.`
+    const confirmLeaveQueue = (queueStatus: 'open' | 'closed', meetingStatus: MeetingStatus | undefined) => {
+        const dialogParts = (meetingStatus !== undefined && meetingStatus === MeetingStatus.STARTED)
+            ? {
+                title: 'Cancel Meeting?',
+                action: 'cancel the meeting',
+                gerund: 'cancelling the meeting',
+                consequences: 'you will no longer meet with the queue host(s)'
+            } : {
+                title: 'Leave Queue?',
+                action: 'leave the queue',
+                gerund: 'leaving the queue',
+                consequences: 'you will lose your place in line'
+            };
+
+        const description = (
+            `Are you sure you want to ${dialogParts.action}? ` +
+            `By ${dialogParts.gerund}, ${dialogParts.consequences}. ` +
+            'If you change your mind, you will have to re-join at the end of the line.' +
+            (queueStatus === 'closed' ? ' The queue is currently closed. You will not be able to re-join until the queue is re-opened.' : '')
```

Comment:
I like the way you adjusted this to work for all the possible scenarios. The only thing I think we should change is that the flow of saying "If you change your mind, you'll have to rejoin. The queue is closed. You can't rejoin." feels a little weird to me in that you're saying, "you have to do this thing but actually you can't and it's slightly weird that I'm not addressing that in the first statement." I think that simply adding "Note:" in front of "The queue is currently closed" might help that feel less weird. We might even make that part Italic. So something like this:

```suggestion
            (queueStatus === 'closed' ? ' <em>Note: The queue is currently closed. You will not be able to re-join until the queue is re-opened.</em>' : '')
```

Reply:
@mfldavidson, are the italics important to you? I'm not absolutely sure if that's possible here. The package we're using might not support HTML or other mark-up within the dialog body. https://www.npmjs.com/package/react-bootstrap-dialog#showoptions

Reply:
What you suggested just ends up being literal tags, since it's part of a string.

Reply:
I'm fine either way, it was just an example.

Reply:
Might be possible to do this, but I'm going to put that off for now, since it'd require some more extensive changes.